build/pipelines/templates/build-single-architecture.yaml

```diff
@@ -38,10 +38,8 @@ jobs:
     fetchDepth: 1
 
   - ${{ if eq(variables['Build.Reason'], 'PullRequest') }}:
```

Comment:
This build task is automatically injected into Microsoft internal pipelines, so we only need to add this task for CI pipeline. Suggest checking `isOSSBuild` for CI pipeline.

Reply:
`isOSSBuild` seems a good condition to me as well. 
Changed.